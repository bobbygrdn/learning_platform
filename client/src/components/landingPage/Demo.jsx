import React from 'react';
import ReactPlayer from 'react-player';

export default function Demo({ video }) {
    return (
        <>
            <ReactPlayer url={video} controls={true} />
        </>
    )
}
