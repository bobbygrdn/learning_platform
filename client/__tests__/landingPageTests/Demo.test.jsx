import React from 'react';
import { render, screen } from '@testing-library/react';
import Demo from '../../src/components/landingPage/Demo';

it('renders without crashing', () => {
    const { container } = render(<Demo />);
    expect(container).toMatchSnapshot();
});

it('should render a react player that plays the demo video', async () => {
    render(<Demo />);
    expect(await screen.findAllByTestId('demo-video')).toHaveLength(1);
});
