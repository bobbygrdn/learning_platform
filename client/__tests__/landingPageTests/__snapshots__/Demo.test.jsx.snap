// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    data-testid="demo-video"
    style="width: 640px; height: 360px;"
  />
</div>
`;
